src/pages/ContentScripts/features/repo-activity-openrank-trends.tsx

```diff
@@ -22,29 +20,24 @@ async function init(): Promise<void> {
   newBorderGridRow.appendChild(newBorderGridCell);
 
   await elementReady('div.Layout-sidebar');
-  render(<RepoActORTrendView currentRepo={repoName} />, newBorderGridCell);
+  render(<RepoActORTrendView currentRepo={getRepoName()} />, newBorderGridCell);
 
   const borderGridRows = $('div.Layout-sidebar').children('.BorderGrid');
   borderGridRows.append(newBorderGridRow);
-
-  // TODO this listener should be deinited once we have prepared a deinit mechanism
-  // deduplicate can avoid rerunning the init() function after browser forward/back,
-  // but echarts will be empty though the dom exists in the page if we do nothing.
-  // so the code below is to render the view to the mount node again as we do before.
-  // it can be improved by extracting the API request code out to avoid redundant request.
-  document.addEventListener('turbo:render', () => {
-    if ($(`#${featureId}`).length > 0) {
-      render(
-        <RepoActORTrendView currentRepo={repoName} />,
-        $(`#${featureId}`).children('.BorderGrid-cell')[0]
-      );
-    }
-  });
 }
 
-void features.add(featureId, {
+const restore = () => {
+  if (exists(`#${featureId}`)) {
+    render(
+      <RepoActORTrendView currentRepo={getRepoName()} />,
+      $(`#${featureId}`).children('.BorderGrid-cell')[0]
+    );
+  }
+};
+
+features.add(featureId, {
   include: [pageDetect.isRepoHome],
   awaitDomReady: false,
-  deduplicate: `#${featureId}`,
   init,
+  restore,
 });
```

Comment:
In feature file, we just need to wrap all restoration code in `restore` and put it as one of the arguments of `features.add()` if needed.